lib/components/channel_panel.dart

```diff
@@ -234,10 +240,59 @@ class _ChannelPanelWidgetState extends State<ChannelPanelWidget> {
           ]),
         );
       }),
+      Consumer<CommandsModel>(builder: (context, commandsModel, child) {
+        return _buildCommandsShortcutBar(commandsModel);
+      }),
       _buildEmotePicker(context)
     ]);
   }
 
+  List<TextButton> _commandButtonsBuilder(CommandsModel commandsModel) {
+    List<TextButton> commandButtons = [];
+    final channelsModel = Provider.of<ChannelsModel>(context, listen: false);
+    for (String command in commandsModel.commands) {
+      commandButtons.add(TextButton(
+          child: Text(command),
+          onPressed: () {
+            ActionsAdapter.instance
+                .send(channelsModel.subscribedChannels.first, command);
+            commandsModel.addCommand(command);
+            _chatInputFocusNode.unfocus();
+          }));
+    }
+    commandButtons.add(
+      TextButton(
+          child: const Text('Clear'),
+          onPressed: () {
+            commandsModel.clear();
+            _chatInputFocusNode.unfocus();
+          },
+          style: TextButton.styleFrom(
+            primary: Colors.red,
+            textStyle: const TextStyle(fontWeight: FontWeight.bold),
+          )),
+    );
+    return commandButtons;
+  }
+
+  Widget _buildCommandsShortcutBar(CommandsModel commandsModel) =>
```

Comment:
I think we should inline these functions. Splitting them up like this is makes the reading order not match the layout order. To reduce the nesting, make a `CommandButtonWidget` or similar. This also makes it easier to test if you so choose :)